src/stats/actions.ts

```diff
@@ -1,59 +1,61 @@
 // @flow
 import _ from 'lodash';
-import SlippiGame from "../index";
 import {
-  States, iterateFramesInOrder
+  State, iterateFramesInOrder
 } from "./common";
 
-import type { ActionCountsType } from "./common";
+import { ActionCountsType } from "./common";
+import { SlippiGame } from '../SlippiGame';
 
-function isRolling(animation) {
-  const rollAnimations = {
-    [States.ROLL_BACKWARD]: true,
-    [States.ROLL_FORWARD]: true,
-  };
-
-  return rollAnimations[animation];
+function isRolling(animation: State): boolean {
```

Comment:
The original code lacked clear typings, since it's supposed to return a boolean for but actually returns either `true` or undefined`.

```
function isRolling(animation) {
const rollAnimations = {
    [States.ROLL_BACKWARD]: true,
    [States.ROLL_FORWARD]: true,
};
return rollAnimations[animation];
}
```
Since you get the benefit of `enum` types in Typescript, it's clearer for the function type signature to specifically be `isRolling(animation): boolean`

Reply:
I see, in that case the old code could be used but instead it could read:
```
  const rollAnimations = {
    [States.ROLL_BACKWARD]: true,
    [States.ROLL_FORWARD]: true,
  };

  return rollAnimations[animation] || false;
```

Reply:
Yes but I think that's much less readable than a simple switch-case statement. Not to mention that you need to instantiate a new object only to return one of the fields, and then to have to `|| false` in order to cast it to a boolean.

Reply:
Very little in this case, I don't mind if you leave it. It's just a pattern I like cause I actually do find it more readable in more complex cases.

Reply:
I also view optimizing for not instantiating objects and avoiding casts as micro-optimizations that are sort of irrelevant, compared to the stuff that actually slows down execution, it's worth the improved readability in my mind. I personally find it pretty readable, to me it reads like:

1. These are the roll animations
2. Is the current animation a roll animation? If not, false

But like I said, for the simple case both are pretty equivalent to me even from a readability standpoint.

Reply:
I think it would make more sense to store it as a list then makes more sense as opposed to an object. Like:
```
const rollAnimations = [States.ROLL_BACKWARD, States.ROLL_FORWARD];
return rollAnimations.contains(animation)
```

Reply:
Once again, not a big deal with a list with 2 objects, but contains is an O(n) lookup. Anyway, we can keep the switch statement.